lib/compiler.js

```diff
@@ -440,7 +442,11 @@ function resolveMiddlewarePath(rootDir, middleware) {
   }
 
   if (!fragment) {
-    resolved.sourceFile = require.resolve(pathName);
+    resolvedPath = resolveAppPath(rootDir, middlewarePath);
+    if (resolvedPath === undefined) {
+      throw new Error(pathName);
```

Comment:
Please provide a descriptive error message that makes it easier for the user to understand what is the problem. See https://github.com/joyent/node/blob/9b534e2e87489d219f0d0cf5be7f9119704fe424/lib/module.js#L337-L341 for an example:

```js
  var filename = Module._findPath(request, paths);
  if (!filename) {
    var err = new Error("Cannot find module '" + request + "'");
    err.code = 'MODULE_NOT_FOUND';
    throw err;
  }
  return filename;
```

Please make the change consistently in all places. Perhaps you can create a method `resolveAppPathOrThrow` that can be used everywhere and keep the code building the exception in one place.